.travis.yml

```diff
@@ -2,15 +2,21 @@ language: cpp
 compiler:
     - gcc
 env:
-    matrix:
-        - FIREFOXVERSION="36.0.1"
-        - FIREFOXVERSION="31.5.0esr"
+    - FX_PATH_SUFFIX=""
+    - FX_PATH_SUFFIX="-esr"
+    - FX_PATH_SUFFIX="-beta"
+#    - FX_PATH_SUFFIX="-prior-esr"
+matrix:
+    fast_finish: true
+    allow_failures:
+        - env: FX_PATH_SUFFIX="-beta"
+        - env: FX_PATH_SUFFIX="-prior-esr"
 notifications:
     email: false
 before_install:
     - export DISPLAY=:99.0
     - sh -e /etc/init.d/xvfb start
-    - wget http://ftp.mozilla.org/pub/firefox/releases/${FIREFOXVERSION}/linux-x86_64/en-US/firefox-${FIREFOXVERSION}.tar.bz2
-    - tar -xjf firefox-${FIREFOXVERSION}.tar.bz2
+    - wget ftp://ftp.mozilla.org/pub/mozilla.org/firefox/releases/latest${FX_PATH_SUFFIX}/linux-x86_64/en-US/firefox-*.tar.bz2
```

Comment:
Best I can tell from reading the docs and related issues on github, at this point there is no HTTP caching on Travis (.com or .org). I've actually never seen it cache the download in all previous builds.

There is APT caching, which is currently only available for .com. Unfortunately, while I was able to get APT installation working for stable, beta, and auroras, the PPAs do not seem to be kept up to date as strictly (especially for aurora). Still seems ok for the current setup, so we could go that route. As I said, though, it's not yet enabled on .org.